src/main/java/org/minions/devfund/carlosorellana/Squarelotron.java

```diff
@@ -0,0 +1,140 @@
+        }
+    }
+
+    /**
+     * Builds new array in base a Squarelotron.
+     *
+     * @return New array with Squarelotron values.
+     */
+    public int[] buildArray() {
+        int[] newArray = new int[this.size * this.size];
+
+        for (int x = 0; x < this.size; x++) {
+
+            for (int y = 0; y < this.size; y++) {
+                newArray[this.size * x + y] = squarelotron[x][y];
+            }
+        }
+
+        return newArray;
+    }
+
+    /**
+     * Finds the Maximum Rings number.
+     *
+     * @return Numbers maximum of rings.
+     */
+    public int findMaxRings() {
+        return (this.size % 2 == 0)
+                ? size / 2
+                : size / 2 + 1;
+    }
+    /**
+      * Modifies A Upside-Down Flip of the outer ring of Squarelotron.
+      *
+      * @param ring - Number of ring.
+      * @return Instance new Squarelotron.
+      */
+    public Squarelotron upsideDownFlip(int ring) {
+
+        if (ring > this.findMaxRings() || ring < 1) {
+            throw new NumberFormatException();
+        }
+
+        Squarelotron squarelotron = new Squarelotron(this.size);
+
+        int first = ring - 1;
+        int last = size - ring;
+
+        for (int i = 0; i <= size - 1; i++) {
+            for (int j = 0; j <= size - 1; j++) {
+                if (i == first || i == last) {
+                    if (j >= first && j <= last) {
+                        squarelotron.squarelotron[i][j] = this.squarelotron[size - 1 - i][j];
+                    }
+                }
+
+                if (i > first && i < last) {
+                    if (j == first || j == last) {
+                        squarelotron.squarelotron[i][j] = this.squarelotron[size - 1 - i][j];
+                    }
+                }
+            }
+        }
+        return squarelotron;
+    }
+
+    public Squarelotron mainDiagonarlFlip(int ring) {
+
+        if (ring > this.findMaxRings() || ring < 1) {
+            throw new NumberFormatException();
+        }
+
+        Squarelotron squarelotron = new Squarelotron(this.size);
+
+        int first = ring - 1; // 0
+        int last = size - ring; // 3
+
+        for (int i = 0; i <= size - 1; i++) {
+            for (int j = 0; j <= size - 1; j++) {
+
+                if (i == first || i == last) {
+                    if (j >= first && j <= last) {
+                        squarelotron.squarelotron[j][i] = this.squarelotron[i][j];
+                    }
+                }
+
+                if (i > first && i < last) {
+                    if (j == first || j == last) {
+                        squarelotron.squarelotron[j][i] = this.squarelotron[i][j];
+                    }
+                }
+            }
+        }
+        return squarelotron;
+    }
+
+    /**
+     * Prints the Squarelotron matrix.
+     *
+     * @return String with the Squarelotron matrix.
+     */
+    @Override
+    public String toString() {
+        StringBuffer sqString = new StringBuffer();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)